[refactor] split login process

rymiyamoto opened the pull request:
I don't think it's good to use goot statements.
For that reason, we reviewed the login process.

Please review.

frankdejonge commented:
What was the problem with the goto statement? What concrete problem are you fixing with this PR?

rymiyamoto commented:
Thank you for your reply.

I think that using goto statement is because the process of attempting to login is complicated.
It's not bad to use a goto statement, but it's not common, so I just split the process and refactored.

Also, the use of goto statements is likely to be complicated in the future.
